pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -544,6 +549,33 @@ type AdmissionPlugin struct {
 	Config *runtime.RawExtension `json:"config,omitempty" protobuf:"bytes,2,opt,name=config"`
 }
 
+// WatchCacheSizes contains configuration of the API server's watch cache sizes.
+type WatchCacheSizes struct {
+	// Default configures the default watch cache size of the kube-apiserver
+	// (flag `--default-watch-cache-size`, defaults to 100).
+	// See: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/
+	// +optional
+	Default *int32 `json:"default,omitempty" protobuf:"varint,1,opt,name=default"`
+	// Resources configures the watch cache size of the kube-apiserver per resource
+	// (flag `--watch-cache-sizes`).
+	// See: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/
+	// +optional
```

Comment:
While its nice that we expose it, the semantics here might be vague. Like when should I as a user decided to set this optional field?

Reply:
Hm, if you want to configure the `--watch-cache-sizes` flag, then you set this list (as stated in the doc string). Why is it vague from your point of view/what could be improved?

Reply:
Point is, why would I want to configure `--watch-cache-sizes`, how will the user see the need to configure this field?

Reply:
Hm, okay, I see what you are meaning. Doesn't this rather belong in k/k documentation? We also don't explain why/when a user should set the OIDC or the feature gates or similar.

Reply:
OIDC is a feature and it's optional (not affecting performance or costs). This feature is managed, either we un-expose  it (because only us understand the problem it tries to solve) or hide it (only operators set it when they see symptoms of cache size issues), or provide documentation on what to do there are 1) unexpected high costs, 2) performance degradation or error logs that are okay (according to the community https://github.com/kubernetes/kubernetes/issues/85723#issuecomment-559583963) but not okay when they happen too often because they signal an underlying problem.

Reply:
I agree, but again, I see this documentation rather in k/k as the kube-apiserver exposes this flag. We just "tunnel it through". I'm happy to refer to such documentation from here, but I don't think we should maintain it ourselves.

Reply:
The issue does not usually happen (its mainly our design and choice of GWs that led it to happen), so kinda felt that we are partially responsible to document the semantics. But sure, was just voicing it out for discussion :)

Reply:
I don't quite get what you mean. This PR is just exposing the two API server flags to make it configurable by users, nothing more.
So this PR alone does not tackle a specific issue, but rather augments the configuration options.
We will of course be using this option to tackle the issue we have seen on big Seeds, but still endusers can run into similar scalability issues, when running large clusters, so it makes sense to have it configurable.

Reply:
> but still endusers can run into similar scalability issues, when running large clusters, so it makes sense to have it configurable. 

This is what I mean, documenting when do they need to use these flags. Lets take this offline if you still don't see my pointl

Reply:
So you are basically asking for a few sentences like in the PR description to be added to the doc string, right?

> Configuring these flags might be useful for large-scale Shoot clusters with a lot of parallel update requests and a lot of watching controllers (e.g. large shooted Seed clusters).
When the API server's watch cache's capacity is too small to cope with the amount of update requests and watchers for a particular resource, it might happen that controller watches are permanently stopped with too old resource version messages.

If that's what you meant, I can do so :)

Reply:
the doc string is good enough for now I guess :)

Reply:
Great, done :)